hawkeye-api/src/handlers.rs

```diff
@@ -363,195 +425,101 @@ pub async fn get_video_frame(id: String, client: Client) -> Result<impl warp::Re
+        WatcherStartStatus::NotFound => ("Watcher can not be found.".to_owned(), StatusCode::OK),
+        WatcherStartStatus::AlreadyRunning => {
+            ("Watcher is already running".to_owned(), StatusCode::OK)
         }
-    };
-
-    // Actions and guards based on the current Watcher status.
-    match deployment.get_watcher_status() {
-        Status::Running => Ok(reply::with_status(
-            // No op, already running!
-            reply::json(&json!({
-                "message": "Watcher is already running"
-            })),
-            StatusCode::OK,
-        )),
-        Status::Pending => Ok(reply::with_status(
-            // No op, committing other changes.
-            reply::json(&json!({
-                "message": "Watcher is currently updating"
-            })),
+        WatcherStartStatus::CurrentlyUpdating => (
+            "Watcher is currently updating".to_owned(),
             StatusCode::CONFLICT,
-        )),
-        Status::Ready => {
-            // Start Watcher by setting Kubernetes deployment replicas=1
-            let patch_params = PatchParams {
-                field_manager: Some("hawkeye_api".to_string()),
-                ..Default::default()
-            };
-
-            // Set Kubernetes deployment replica=1 via patch.
-            let deployment_scale_json = json!({
-                "apiVersion": "autoscaling/v1",
-                "spec": { "replicas": 1 },
-            });
-            deployments_client
-                .patch_scale(
-                    deployment.metadata.name.as_ref().unwrap(),
-                    &patch_params,
-                    &Patch::Merge(&deployment_scale_json),
-                )
-                .await
-                .unwrap();
-
-            // Update the status of the Watcher to indicate it should be running.
-            let status_label_json = json!({
-                "apiVersion": "apps/v1",
-                "metadata": {
-                    "labels": {
-                        "target_status": Status::Running,
-                    }
-                }
-            });
-            deployments_client
-                .patch(
-                    deployment.metadata.name.as_ref().unwrap(),
-                    &patch_params,
-                    &Patch::Merge(status_label_json),
-                )
-                .await
-                .unwrap();
-
-            Ok(reply::with_status(
-                reply::json(&json!({
-                    "message": "Watcher is starting"
-                })),
-                StatusCode::OK,
-            ))
-        }
-        Status::Error => Ok(reply::with_status(
-            reply::json(&json!({
-                "message": "Watcher in error state cannot be set to running"
-            })),
+        ),
+        WatcherStartStatus::InErrorState => (
+            "Watcher in error state cannot be set to running".to_owned(),
             StatusCode::NOT_ACCEPTABLE,
-        )),
-    }
+        ),
+        _ => {
```

Comment:
I suggest not having this wildcard, in case new enum variants are added in the future...

```suggestion
        WatcherStartStatus::Starting => {
```

Reply:
Ah, good call. I should be using the success state here anyways.